lib/Main.js

```diff
@@ -180,7 +180,9 @@ Main.prototype.putUserToStore = function(user) {
 };
 
 Main.prototype.getUrlForMxc = function(mxc_url) {
-    return this._config.homeserver.url + "/_matrix/media/v1/download/" +
+    const hs = this._config.homeserver;
+    const url = hs.media_url !== undefined ? hs.media_url : hs.url;
```

Comment:
can probably just get away with `hs.media_url || hs.url` or `hs.media_url ? hs.media_url : hs.url` if you're worried about accidental booleans.